processes/processes.tex

```diff
@@ -2,196 +2,229 @@ \chapter{Processes}
-\item A set of machine instructions
-\item A number denoting which instruction to start from
-\item Constants
-\item Libraries to link and where to fill in the address of those libraries
+    \item A binary format: This tells the operating system about the various sections of bits in the binary -- which parts are executable, which parts are constants, which libraries to include etc.
+    \item A set of machine instructions
+    \item A number denoting which instruction to start from
+    \item Constants
+    \item Libraries to link and where to fill in the address of those libraries
 \end{itemize}
 
 Processes are very powerful but they are isolated!
+
 That means that by default, no process can communicate with another process.
-This is very important because if you have a large system (let's say the University of Illinois Engineering Workstations) then you want some processes to have higher privileges than your average user, and one certainly doesn't want the average user to be able to bring down the entire system either on purpose or accidentally by modifying a process.
-As most of you have realized by now if you stuck the following code snippet into a program, the variables would not be shared between two parallel invocations of the program.
+
+This is very important because if you have a large system (let's say the University of Illinois Engineering Workstations (EWS)) then it is likely that you want some processes to have higher privileges than your average user. One certainly doesn't want the average user to be able to bring down the entire system, either on purpose or by accidentally by modifying a process.
+As most of you have realized by now, if you stuck the following code snippet into a program, the variables would not be shared between two parallel invocations of the program.
 
 \begin{lstlisting}[language=C]
 int secrets;
 secrets++;
 printf("%d\n", secrets);
 \end{lstlisting}
 
-On two different terminals, as you would guess they would both print out 1 not 2.
-Even if we changed the code to do something really hacky, there would be no way to change another process' state unintentionally.
-There are ways of changing it purposefully though.
+On two different terminals, they would both print out 1 not 2.
+Even if we changed the code to attempt to affect other process instances, there would be no way to change another process' state unintentionally.
+However, there are other intentional ways to change the program states of other processes.
 
 \section{Process Contents}
 
 \subsection{Memory Layout}
 
-When a process starts, it gets its own address space. Each process gets the following.
+When a process starts, it gets its own address space. Each process gets the following:
 
 \begin{itemize}
 \item \textbf{A Stack}.
-The stack is the place where automatic variable and function call return addresses are stored.
+The stack is the place where automatically allocated variables and function call return addresses are stored.
```

Comment:
we should introduce the terminology `automatic variables` somewhere here as well.